neutron/include/scipp/neutron/constants.h

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include <boost/units/systems/si/codata/electromagnetic_constants.hpp>
+#include <boost/units/systems/si/codata/neutron_constants.hpp>
+#include <boost/units/systems/si/codata/universal_constants.hpp>
+
+#include "scipp/common/constants.h"
+
+#include "scipp/units/unit.h"
+
+#include "scipp/variable/operations.h"
+
+#include "scipp/neutron/beamline.h"
+
+namespace scipp::neutron::constants {
```

Comment:
Since this namespace is holding both physical constants and conversion functions, maybe we can find a more generally encompassing name?
Especially since you have something called `conversions` just below, i think it's a bit confusing.
Or maybe it's simply the function names that are confusing?

But when I see `tof_to_dspacing` in one file/namespace, and `tof_to_energy` in another file/namespace, i'm wondering why they are not in the same place, as I get the feeling they should be doing similar things.

Reply:
The namespace only contains conversion "constants", *not*, the functions (those are in `conversions`). Here "constant" is seen as constant for every pixel, i.e., these return the array of constants used in equations for unit conversion.

This is essentially just an implementation detail of the unit conversion. The only reason for being in a header is for testing.

Reply:
yes I understand that it's just constants, I just thought having the same name for both the constant and the function was a bit confusing

Reply:
I thought the opposite? Aren't they distinguished by their namespace name? If also the name would differ?

Reply:
or put another way: you have variables called `conversionFactor` in a `constants` namespace, while you also have a `conversions` namespace.

Reply:
So you are saying I should rename the `conversions` namespace?

Reply:
It's not super important.
When I look again at the syntax above
```
return convert_generic(std::move(d), from, to, conversions::tof_to_energy,
                           constants::tof_to_energy(d));
```
it's actually ok to have it like that. I see your point where it's convenient to have the same name.

And we can always change it later if we really want to.